test/rules/prefer_typing_uninitialized_variables.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N prefer_typing_uninitialized_variables`
+
+class BadClass {
+  static var bar; // LINT
+  var foo; // LINT
+  final baz; // LINT
+  var a = 5,
+      b; // LINT
+  String d;
+
+  BadClass(this.baz);
+
+  void method() {
+    var bar; // LINT
+    bar = 5;
+    print(bar);
+  }
+
+  void someMethod() {
+    for (final v in <String>[]) {
```

Comment:
Wrong style of for loop :-). I should have been more explicit. (I usually call this a for-in loop to distinguish between the two.) VariableDeclarationList isn't a child of ForInStatement because can declare at most one variable. But that isn't the case in the "normal" for loop:
```
for (var i, j = 0; j < 5; i = j, j++) {
  print(i);
}
```